packages/icons/src/constants.ts

```diff
@@ -9,24 +9,43 @@ export const ID_UNWANTED_PARTS = ['shape'];
 export const ID_SYMBOL_PREFIX = 's';
 export const ID_SYMBOL_SEPERATOR = '';
 
-export const ICON_SIZE_VAR_NAME = 'pis';
-
 export const ICON_V1_TEMPLATE = `<svg xmlns="http://www.w3.org/2000/svg">
   <defs>
     {symbols}
   </defs>
-
   {uses}
 </svg>`;
 
+const ICON_V2_TEMPLATE_STYLES = `<style>
+  ${ICON_V2_SIZES.map((size, i) => {
+    const isFirst = size === ICON_V2_SIZES[0];
+    const isLast = size === ICON_V2_SIZES[ICON_V2_SIZES.length - 1];
+    const nextSize = ICON_V2_SIZES[i + 1];
+    const min = !isFirst && `(min-width: ${size}px)`;
+    const max = !isLast && `(max-width: ${nextSize - 1}px)`;
+
+    return `@media ${[min, max].filter(Boolean).join(' and ')} {
+      g {
+        --${ID_SYMBOL_PREFIX}${size}: block;
+      }
+    }`;
+  }).join('\n\n')}
```

Comment:
This function puzzled me quite a bit and just for a mental exercise I added a suggestion for making it a little bit more concise. While I like the shorthand boolean filter, this version does not need it. If you decide to not implement it, consider subtracting only 0.01 px from the max-width media query to catch edge cases where subpixel rendering would end up between two queries and render the mobile (?) icon unexpectedly.

```suggestion
const ICON_V2_TEMPLATE_STYLES = `<style>
  ${ICON_V2_SIZES.map((size, i) => {
    const query = [];
    if (i > 0) {
      query.push(`(min-width: ${ICON_V2_SIZES[i]}px)`);
    }
    if (i < ICON_V2_SIZES.length - 1) {
      query.push(`(max-width: ${ICON_V2_SIZES[i + 1] - 0.01}px)`);
    }

    return `@media ${query.join(' and ')} {
      g {
        --${ID_SYMBOL_PREFIX}${size}: block;
      }
    }`;
  }).join('\n\n')}
```